temporal-sdk/src/main/java/io/temporal/internal/testservice/TestWorkflowService.java

```diff
@@ -259,10 +260,14 @@ StartWorkflowExecutionResponse startWorkflowExecutionImpl(
         }
       }
       Optional<TestServiceRetryState> retryState;
+      Optional<Failure> lastFailure = Optional.empty();
```

Comment:
I would move empty() initialization to else block:

Optional<Failure> lastFailure;
if (...) {
   lastFailure = ...
} else {
   lastFailure = Optional.empty();
}

Reply:
This would require us to set it to empty in two branching paths to ensure it's non-null, thus more verbose. Any particular reason not to leave it this way?